tradingeconomics/adapter.js

```diff
@@ -0,0 +1,99 @@
+const { Requester, Validator } = require('@chainlink/external-adapter')
+const TeClient = require('tradingeconomics-stream')
+const API_CLIENT_KEY = process.env.API_CLIENT_KEY
+const API_CLIENT_SECRET = process.env.API_CLIENT_SECRET
+const DEFAULT_WS_TIMEOUT = 5000
+const wsTimeout = process.env.WS_TIMEOUT || DEFAULT_WS_TIMEOUT
+
+const customParams = {
+  base: ['base', 'from', 'asset']
+}
+
+const commonSymbols = {
+  N225: 'NKY:IND',
+  FTSE: 'UKX:IND'
+}
+
+const createRequest = (input, callback) => {
+  const validator = new Validator(callback, input, customParams)
+  const jobRunID = validator.validated.id
+  let symbol = validator.validated.data.base.toUpperCase()
+  if (symbol in commonSymbols) {
+    symbol = commonSymbols[symbol]
+  }
+
+  const client = new TeClient({
```

Comment:
We probably do not need to recreate this client on every request, so we can move it outside of `createRequest` function.

Reply:
Since external adapters are commonly ran on serverless frameworks, we have to assume that we don't retain state between requests.

Reply:
I do not think there should be a problem because of serverless deployment. The server is either active between function calls, which means it retains state. Or it's teared-down and built back up (cold run) on next function call - that means the state is re-initialized before the function is run.

But this is nothing critical so let's leave it for now.